src/interface/src/app/map/map-control-panel/condition-tree/condition-tree.component.ts

```diff
@@ -50,29 +58,76 @@ export class ConditionTreeComponent implements OnInit {
   );
 
   treeControl = new FlatTreeControl<ConditionFlatNode>(
-    (node) => node.level,
-    (node) => node.expandable
+    (node) => node!.level,
+    (node) => node!.expandable
   );
   conditionDataSource = new MatTreeFlatDataSource(
     this.treeControl,
     this.treeFlattener
   );
 
-  constructor() {}
+  public refreshTree() {
 
+    var _data = this.conditionDataSource.data;
+    this.treeFlattener = new MatTreeFlattener(
+      this._transformer,
+      (node) => node.level,
+      (node) => node.expandable,
+      (node) => node!.children
+    );
+  
+    this.treeControl = new FlatTreeControl<ConditionFlatNode>(
+      (node) => node!.level,
+      (node) => node!.expandable
+    );
+    this.conditionDataSource = new MatTreeFlatDataSource(
+      this.treeControl,
+      this.treeFlattener,
+    );
+    this.conditionDataSource.data = _data;
+    this.cdr.detectChanges();
+  }
+
+  constructor(private cdr: ChangeDetectorRef) {}
+  
   ngOnInit(): void {
-    this.conditionsData$
-      .pipe(
-        filter((data) => data.length > 0),
-        take(1)
-      )
-      .subscribe((data) => {
-        this.conditionDataSource.data = data;
-        // Ensure the radio button corresponding to the saved selection is selected.
-        this.map.config.dataLayerConfig = this.findAndRevealNode(
+    if(this.dataType == ConditionTreeType.RAW){
```

Comment:
nit: the spacing here and around the file should be made consistent with the existing code in the file.